src/kafka-tests/Unit/FakeProtocolGatewayTest.cs

```diff
@@ -0,0 +1,323 @@
+﻿using kafka_tests.Helpers;
+using KafkaNet;
+using KafkaNet.Model;
+using KafkaNet.Protocol;
+using Moq;
+using Ninject.MockingKernel.Moq;
+using NUnit.Framework;
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace kafka_tests.Unit
+{
+    [TestFixture]
+    [Category("unit")]
+    public class FakeProtocolGatewayTest
```

Comment:
Weird name. Is the protocol gateway fake? if it is, you're testing something else and the name should reflect that. If it isn't fake, the name shouldn't say that.